src/boot/efi/pe.c

```diff
@@ -206,6 +216,11 @@ static void pe_locate_sections(
                                         continue;
                         }
 
+                        /* Special handling for .dtbauto sections compared to plain .dtb */
+                        if (validate_base && pe_section_name_equal(section_names[i], ".dtbauto"))
+                                if (!pe_use_this_dtb(section_table, n_section_table, SIZE_TO_PTR(validate_base), j->VirtualAddress, j->VirtualSize, i))
+                                        continue;
```

Comment:
i think the whole function should fail early if .dtbauto is requested but "validate_base" not provided.

Reply:
Hopefully fixed in the latest revision. Now if .dtbauto is requested but no `validate_base` is provided no sections would be matched.